command-line/command-line-interface.md

```diff
@@ -131,6 +131,48 @@ We tend to use kebab-case for directory (folder) names and git repository names.
 
 7. Navigate back into the "foo" directory. Run `mv second.txt foofile.txt`, then run `ls`. What happened to "second.txt"?
 
+8. Use `TAB key` to autocomplete the names of directories and files while in the command line faster as it'll autocomplete the string as far as it can before you have to disambiguate. 
+    Hit the tab twice to see the list of AUTOCOMPLETE POSSIBILITES (if multiple matches).
+
+9. 	The `root directory` is the directory that contains all other directories and files on the system and which is designated by a forward slash ( / ). `root` is the very top directory of the directory tree diagram.
+    Root directory of a github project is the top most directory of your git project which contains all the files hosted on github.
+
+10. You can immediately open whatever folder or directory you are working within into the Finder of MacOS and Mac OS X by simply typing `open .` and executing it   
+   `open` is not magically connected to Mac's Finder, but actually it's just using whatever default program on your Mac opens a file with that extension, and in the case of directories, the default program is Finder.
+
+11. Another mac shortcut `cmd+ shift+ g` can be used anywhere you see a Finder window (even in other Mac programs such as sourcetree or vs code when you're opening a file or directory). This brings up a little text box into which you can type or paste a path. Use pwd to get your current path, then copy it from the Terminal and open up sourcetree, go to file > open, then use `cmd + shift + g` to paste in the path you want.    
+
+12. `ls` command is used to peek into other folders.
+       The list of ls argumnets are
+          ls -l : shows file or directory, size, modified date and time, file or folder name and owner of file and its permission.
+          ls -a : lists all the files including hidden files.
+          ls -lh: shows sizes in human readable format.
+
+### cd commands
```

Comment:
Please write all commands using back-ticks to distinguish between plain text and commands